### DISCUSSION TITLE
Ability to specify URLS per client

### DISCUSSION THREAD

**Discussion by jifeon** [ory/hydra#3090] [Q&A] [answered]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

We are hosting a lot of sites, they all work on the same platform, they all have their own domain name. We have user/password authentication and it happens on a subpage for every domain. After successful authentication we start storing a token in cookies.

Recently we started to provide authentication for 3rd party systems using accounts on our platform - we become OAuth provider. In order to do this we are using self hosted hydra instance. 

The problem is that you need to specify `URLS_*` env vars per hydra instance. It means that we need to run new hydra instance for every client who needs this feature. Currently it's quite problematic to move global authentication to separate domain and use it for all our sites.

### Describe your ideal solution

In ideal world hydra provides possibility to setup `URLS_*` env vars as settings while client creation, like

```sh
clients create \
  --endpoint http://hydra:4445 --id client-id --secret secret \
  --callbacks https://callback.url \
  --urls-login=https://ourdomain.com/oauth/login \
  --urls-consent=https://ourdomain.com/oauth/consent \
  --urls-self-issuer=https://ourdomain.com/oauth \
  --urls-self-public=https://ourdomain.com/oauth
```

note params started with `--urls`

### Workarounds or alternatives

Currently we run separate hydra instance for every site. Proper solution could be moving authentication to separate domain, which is our long term goal.

### Version

v1.10.2

**Comment by aeneasr** [comment, marked as answer]
We usually recommend to implement the UI switch in your login / consent app. Based on the client ID, which you can get from the flow, you can decide which login UI to show!
